Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs

```diff
@@ -627,7 +637,14 @@ Task<bool> SwitchContentAsync(Page view, bool animated, bool removed = false, bo
 					fragments.Add(fragment);
 				}
 			}
-			transaction.Commit();
+
+			if (Build.VERSION.SdkInt >= BuildVersionCodes.N)
+				transaction.CommitNow();
+			else
+			{
+				transaction.Commit();
+				new Handler(Looper.MainLooper).PostAtFrontOfQueue(() => FragmentManager.ExecutePendingTransactions());
```

Comment:
At the moment, I'm not sure if there are any ramifications of executing pending transactions here. I'd like to think the commits in this renderer should not be eventually executed. Similarly, `CommitNow()` should do the same job.